filament/src/details/Renderer.cpp

```diff
@@ -186,13 +186,21 @@ void FRenderer::render(FView const* view) {
 
     assert_invariant(mSwapChain);
 
-    if (mBeginFrameInternal) {
+    if (UTILS_UNLIKELY(mBeginFrameInternal)) {
+        // this should not happen, the user should not call render() if we returned false from
+        // beginFrame(). But because this is allowed, we handle it gracefully.
         mBeginFrameInternal();
         mBeginFrameInternal = {};
     }
 
     if (UTILS_LIKELY(view && view->getScene())) {
+        if (mViewRenderedCount) {
+            // this is a good place to kick the GPU, since we've rendered a View before
+            // and we're about to render another one.
+            mEngine.getDriverApi().flush();
```

Comment:
Will this be done for every view or just after the first one?

Reply:
This is only done starting with the 2nd view (and obviously not done for the last one -- which is a better way to look at it).